src/plots/segmentPlot.ts

```diff
@@ -0,0 +1,146 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+export module Plots {
+  export class Segment<X, Y> extends XYPlot<X, Y> {
+    private static _X2_KEY = "x2";
+    private static _Y2_KEY = "y2";
+
+    /**
+     * A Segment Plot displays line segments based on the data.
+     *
+     * @constructor
+     */
+    constructor() {
+      super();
+      this.addClass("segment-plot");
+    }
+
+    protected _createDrawer(dataset: Dataset) {
+      return new Drawers.Segment(dataset);
+    }
+
+    protected _generateDrawSteps(): Drawers.DrawStep[] {
+      return [{attrToProjector: this._generateAttrToProjector(), animator: new Animators.Null()}];
+    }
+
+    /**
+     * Gets the AccessorScaleBinding for X
+     */
+    public x(): AccessorScaleBinding<X, number>;
+    /**
+     * Sets X to a constant value or the result of an Accessor.
+     *
+     * @param {X|Accessor<X>} x
+     * @returns {Plots.Segment} The calling Segment Plot.
+     */
+    public x(x: number | Accessor<number>): Plots.Segment<X, Y>;
+    /**
+     * Sets X to a scaled constant value or scaled result of an Accessor.
+     * The provided Scale will account for the values when autoDomain()-ing.
+     *
+     * @param {X|Accessor<X>} x
+     * @param {Scale<X, number>} xScale
+     * @returns {Plots.Segment} The calling Segment Plot.
+     */
+    public x(x: X | Accessor<X>, xScale: Scale<X, number>): Plots.Segment<X, Y>;
+    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: Scale<X, number>): any {
+      if (x == null) {
+        return super.x();
+      }
+      if (xScale == null) {
+        return super.x(<number | Accessor<number>>x);
+      } else {
+        return super.x(<X | Accessor<X>>x, xScale);
+      }
+    }
+
+    /**
+     * Gets the AccessorScaleBinding for X2
+     */
+    public x2(): AccessorScaleBinding<X, number>;
+    /**
+     * Sets X2 to a constant number or the result of an Accessor.
+     * If a Scale has been set for X, it will also be used to scale X2.
+     * 
+     * @param {number|Accessor<number>|Y|Accessor<Y>} y2
+     * @returns {Plots.Segment} The calling Segment Plot
+     */
+    public x2(x2: number | Accessor<number> | X | Accessor<X>): Plots.Segment<X, Y>;
+    public x2(x2?: number | Accessor<number> | X | Accessor<X>): any {
+      if (x2 == null) {
+        return this._propertyBindings.get(Segment._X2_KEY);
+      }
+      var xBinding = this.x();
+      var xScale = xBinding && xBinding.scale;
```

Comment:
If the user sets `x2()` before `x()`, they won't get the `Scale`, although I'll note that this is an open bug on `Plots.Rectangle` as well.